components/org.wso2.carbon.identity.oauth.dcr/src/main/java/org/wso2/carbon/identity/oauth/dcr/service/DCRManagementService.java

```diff
@@ -312,7 +313,197 @@ public void unregisterOAuthApplication(String userId, String applicationName, St
+            .setTenantDomain(tenantDomain, true);
+        PrivilegedCarbonContext.getThreadLocalCarbonContext().setUsername(userName);
+
+        ApplicationManagementService appMgtService = DCRDataHolder.getInstance().
+            getApplicationManagementService();
+        if (appMgtService == null) {
+          throw new IllegalStateException("Error occurred while retrieving Application "
+              + "Management Service");
+        }
+
+        OAuthAdminService oAuthAdminService = new OAuthAdminService();
+        OAuthConsumerAppDTO existingApp;
+
+        try {
+          existingApp = oAuthAdminService.getOAuthApplicationData(updateRequestProfile
+              .getConsumerKey());
+        } catch (IdentityOAuthAdminException e) {
+          throw new DCRException(e.getMessage(), e);
+        }
+
+        ServiceProvider existingServiceProvider;
+
+        try {
+          existingServiceProvider = appMgtService.getServiceProvider(existingApp
+              .getApplicationName(), tenantDomain);
+        } catch (IdentityApplicationManagementException e) {
+          throw new DCRException(e.getMessage(), e);
+        }
+
+        if (!updateRequestProfile.getClientId().equals(existingApp.getOauthConsumerKey())) {
+          throw new DCRException("The included client ID is not a valid value");
+        }
+
+        if (!updateRequestProfile.getClientSecret().equals(existingApp.getOauthConsumerSecret())) {
+          throw new DCRException("The included client secret is not a valid value");
+        }
+
+        OAuthConsumerAppDTO oAuthConsumerApp = new OAuthConsumerAppDTO();
+        oAuthConsumerApp.setOauthConsumerKey(updateRequestProfile.getClientId());
+        oAuthConsumerApp.setOauthConsumerSecret(updateRequestProfile.getClientSecret());
+        oAuthConsumerApp.setApplicationName(updateRequestProfile.getClientName());
+
+        if (updateRequestProfile.getRedirectUris().size() == 0) {
+          String errorMessage = "RedirectUris property must have at least one URI value.";
+          throw IdentityException.error(DCRException.class,
+              ErrorCodes.META_DATA_VALIDATION_FAILED.toString(), errorMessage);
+        }
+
+        oAuthConsumerApp.setCallbackUrl(updateRequestProfile.getRedirectUris().get(0));
+
+        String grantTypes = StringUtils.join(updateRequestProfile.getGrantTypes(), " ");
+        if (grantTypes.isEmpty()) {
+          grantTypes = GrantTypes.GRANT_TYPE_CODE;
+        }
+
+        oAuthConsumerApp.setGrantTypes(grantTypes);
+        oAuthConsumerApp.setOAuthVersion(OAUTH_VERSION);
+
+        OAuthConsumerAppDTO updatedApp;
+        try {
+          oAuthAdminService.updateConsumerApplication(oAuthConsumerApp);
+          updatedApp = oAuthAdminService.getOAuthApplicationData(updateRequestProfile
+              .getConsumerKey());
+        } catch (IdentityOAuthAdminException e) {
+          throw new DCRException(e.getMessage(), e);
```

Comment:
If you catch and throw a new exception, you have to add a informative error message that is meaningful for this context, rather than just adding the error message from the caught exception.